lib/bandit/http1/adapter.ex

```diff
@@ -29,27 +30,29 @@ defmodule Bandit.HTTP1.Adapter do
     with {:ok, headers, method, request_target,
           %__MODULE__{version: version, buffer: buffer} = req} <-
            do_read_headers(req) do
-      body_size = get_header(headers, "content-length")
       body_encoding = get_header(headers, "transfer-encoding")
       connection = get_header(headers, "connection")
       keepalive = should_keepalive?(version, connection)
 
-      case {body_size, body_encoding} do
-        {nil, nil} ->
+      case {get_content_length(headers), body_encoding} do
+        {{:ok, nil}, nil} ->
```

Comment:
can we use a clause in the above `with` to match on `{:ok, body_size} <- get_content_length(headers)`? I think that would simplify the case matches here (specifically you could drop the nested tuples and also the `{:error...` clause).